### ISSUE TITLE
Remove `uv` and dependencies that are not direct

### ISSUE THREAD

**Issue by llucax** [frequenz-floss/frequenz-dispatch-python#13]
### What's needed?

`uv` is not really being used, so we shouldn't depend on it (developers that want to use it can install it manually on their envs).

The non-direct dependencies should go, I guess they were added because of `uv`?. See:

* https://github.com/frequenz-floss/frequenz-dispatch-python/pull/9#discussion_r1539263157

 I'm OK to keep the `uv` and non-direct dependencies experimentally for a while, but we should at least add a comment to the dependency saying why it is there, and the current issues we have with `uv`. ~Also if it is not much work it would be nice if it were actually used in the CI, so we can also see more in practice how much faster the CI could be with it.~

**Comment by llucax** [comment]
I see it is used in the CI now.

**Comment by Marenz** [comment]
Do you still think we should remove it?

**Comment by llucax** [comment]
Nah, I mean, I think we should move officially to use `uv` sooner than later, it is just weird that this project sits in the middle for now but I wouldn't invest time on this as we know eventually it will need updating anyways.

**Comment by llucax** [comment]
That said, all non-direct dependencies in the discussed pr that are not pre-releases should be removed I think. I guess the only reason to add them was to make it work with uv and it only complains about indirect pre-release dependencies. I think by now we should not use any since we have final releases for dispatch, right?

**Comment by Marenz** [comment]
I think they also changed the behavior, not sure. But it became less of a pain

**Comment by Marenz** [comment]
Closing this issue as the original concerns have been addressed:

- **uv is now actively used** in CI via the `dev-noxfile` dependencies
- **Git-based dependencies are gone** - the problematic dependencies like `frequenz-dispatch-client @ git+...` from PR #9 have been replaced with proper versioned packages
- **`frequenz-client-base` could be removed** - it's a transitive dependency via `frequenz-client-dispatch`, though the code does import from it directly. The explicit pinning may be intentional to constrain versions, but is not strictly required.

As llucax noted in the October 2024 comments, investing time on this isn't worthwhile since the project will eventually move fully to `uv` anyway.

**Comment by llucax** [comment]
> though the code does import from it directly

Then it is a direct dependency.
